### PR TITLE
Basic functionality of Sublime Text plugin for SDL logs.

### PR DESCRIPTION
**Pull request by dkandyba**
This PR is **ready** for review.
**Summary**
This PR provides basic functionality of Sublime Text 3 plugin for working with sdl_core logs:

- basic logs analyse;
- basic color highlight;

**CLA**
☑︎ I have signed the CLA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,25 @@
...
 Sublime Text plugin  for handy work with SDL logs.
+
+##Requirements
+This plugin requires at least Sublime Text 3 v.3176 or higher for best perfomance and correct syntax
+
+## Installing
+- Move directory "SDL Logs" to Sublime Packages folder.
+(Generally at /home/*USER*/.config/sublime-text-3/Packages)
+(Creating folder for plugin in Pacckages is good practise, you can choose any name for it)
+- Necessary open "preferences->Package->Settings->SDL Logs->Setting" and change attribute "source_path" on own path to source sdl_core.
```
Comment (IGapchuk) [inline README.md]: @dkandyba I think, it's not a good idea to create separate variable for source_path. It is redundant. In logs exists absolute path to file. You have to use it.

File: SDL Logs/Main.sublime-menu [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      { "caption": "Package Settings",
+        "mnemonic": "P",
+        "id": "package-settings",
+        "children": [
+          { "caption": "SDL Logs",
+            "children": [
+              { "caption": "README",
+                "command": "open_file",
+                "args": {
+                  "target": "${packages}/SDL Logs/README.md"
```
Comment (IGapchuk) [inline SDL Logs/Main.sublime-menu]: @dkandyba You have to change directory name, without white space in name.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,25 @@
 # sdl_sublime_log_plugin
 Sublime Text plugin  for handy work with SDL logs.
+
+##Requirements
```
Comment (IGapchuk) [inline README.md]: @dkandyba Add a white space between double sharp and "Requirements" word.

File: README.md (hunk 1 above)
Comment (dkandyba) [reply README.md]: @IGapchuk fixed in e03ffdf965eeace46e1143a5edf8b915922b48be

File: SDL Logs/Main.sublime-menu (hunk 2 above)
Comment (dkandyba) [reply SDL Logs/Main.sublime-menu]: @IGapchuk fixed in e03ffdf965eeace46e1143a5edf8b915922b48be

File: README.md (hunk 3 above)
Comment (dkandyba) [reply README.md]: @IGapchuk fixed in e03ffdf965eeace46e1143a5edf8b915922b48be

File: SdlLogs/SdlLogs.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+pathRegex       = "sdl_core/src/(.*).(cc|h):[0-9]{1,5}" 
+
+#flags for commands
+actThreadFlag    = False
+actDateFlag      = False
+actPathFlag      = False
+actComponentFlag = False
+
+def getSelectedText(self):
+	#array of selection regions
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba avoid comments for lines. code should be as much self documented as possible. For improoving self documentation you can choose another variable names.
But do not avoid comments for functions.

File: SdlLogs/SdlLogs.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	selText = self.view.sel()
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba what coding style are you following? I propose to use https://www.python.org/dev/peps/pep-0008/

File: SdlLogs/SdlLogs.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+	#selected string
+	selText = self.view.substr(selText[0])
+	return selText
+
+
+#Hide date of all traces from log
+class HideDateCommand(sublime_plugin.TextCommand):
+	def run(self, edit):
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba propose to use doxygen styles for comments. And describe in comments arguments passed to function. 
Not clear what does `edit` argument means.
https://www.python.org/dev/peps/pep-0257/

File: SdlLogs/SdlLogs.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+			actDateFlag = False
+
+
+class FunctionCallCommand(sublime_plugin.TextCommand):
+	def run(self, edit):
+		posFatal = 0
+		enterRegex = "initial fake value"
+
+		while enterRegex is not "":
+
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba such empty lines are redundant

File: SdlLogs/SdlLogs.py [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+			enterRegion = self.view.find(threadEnter, posFatal)
+			enterRegex = self.view.substr(enterRegion)
+			enterRegex = enterRegex[enterRegex.find(" ")+1:]
+			enterRegex = enterRegex[:enterRegex.find(" ")]
+
+			print (enterRegex)
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba is it debug print? Will it be deleted after some release? If not, please make it some in  understandable format

File: SdlLogs/SdlLogs.py [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+			self.view.unfold(trcRegions)
+			actDateFlag = False
+
+
+class FunctionCallCommand(sublime_plugin.TextCommand):
+	def run(self, edit):
+		posFatal = 0
+		enterRegex = "initial fake value"
+
+		while enterRegex is not "":
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba not clear what is going on in this cycle, Propose to move this cycle to function with appropriate name

File: SdlLogs/SdlLogs.py [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+
+			enterRegion = self.view.find(threadEnter, posFatal)
+			enterRegex = self.view.substr(enterRegion)
+			enterRegex = enterRegex[enterRegex.find(" ")+1:]
+			enterRegex = enterRegex[:enterRegex.find(" ")]
+
+			print (enterRegex)
+
+			if enterRegex is not "":
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba is code inside reachable? Whe previous line is `while enterRegex is not "":` it means that previous cycle won't funish until `enterRegex` won't be "".

File: SdlLogs/SdlLogs.py [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+				exitRegion = self.view.find(enterRegex+threadExit, posFatal)
+				print (self.view.substr(exitRegion))
+
+				if exitRegion.size() is not 0:
+					posFatal = enterRegion.b
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba what is `b` here? bad name. Is it standard name?

File: SdlLogs/SdlLogs.py [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+dateRegex       = "\\[\\s?[0-9]{1,2} \\s?[A-Z][a-z]{2} [0-9]{4} [0-9]{1,2}:[0-9]{1,2}:[0-9]{1,2},[0-9]{0,3}]"
+junkRegex       = " /(.*)/sdl_core/src/" 
+componentRegex  = "\\[[A-Z][a-zA-z]{3,26}] "
+threadEnter     = ".(cc|h):[0-9]{1,5} (.*): Enter"
+threadExit      = " Exit"
+threadRegex     = "\\[0x[a-zA-Z0-9]{12}]"
+pathRegex       = "sdl_core/src/(.*).(cc|h):[0-9]{1,5}" 
+
+#flags for commands
+actThreadFlag    = False
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba is it possible to have several act flags active in time

File: SdlLogs/SdlLogs.py [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+
+			for i in range (0, len(trcRegions)):
+				trcRegions[i] = self.view.full_line(trcRegions[i])
+				trcRegions[i] = self.view.substr(trcRegions[i])
+
+			self.view.erase(edit, sublime.Region(0, self.view.size()))
+			leftBorder = 0
+
+			for i in range (0, len(trcRegions)):
+				leftBorder+= self.view.insert(edit, leftBorder, trcRegions[i])
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba trcRegions used only for reading.
Use range based for across trcRegions:
```
for trc_region in trcRegions:
  leftBorder+= self.view.insert(edit, leftBorder, trc_region)
```

File: SdlLogs/SdlLogs.py [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+				file = self.view.settings().get(sourcePath) + primaryPath.group(2)
+				if path.isfile(file) : 
+					self.view.window().open_file(file + ":" + primaryPath.group(4), sublime.ENCODED_POSITION)
+				else:
+					sublime.error_message("File '{0}' not found. Maybe path to source files in your settings is incorrect or file really absent".format(file))
+			elif isItPath:
+				sublime.error_message("Expected '/sdl_core/src/' in '{0}' ".format(isItPath.group(0)))
+		else:
+			if primaryPath :
+				file = primaryPath.group(1) + primaryPath.group(2)
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py]: @dkandyba this code looks similar. May be extracted to a function?

File: SdlLogs/SdlLogs.sublime-color-scheme [hunk 15]
```diff
@@ -0,0 +1,123 @@
+{
+    "variables": {
+        // Define variables here
+    },
+    "globals": {
+            "background":     "#22282A", 
+            "caret":      "#AEAFAD",
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.sublime-color-scheme]: @dkandyba please perform some formatting of this file

**Review by LuxoftAKutsan — CHANGES_REQUESTED** [review]

File: SdlLogs/SdlLogs.py (hunk 11 above)
Comment (vilyukh) [reply SdlLogs/SdlLogs.py]: @LuxoftAKutsan Yes, it`s a property of Region Class from Sublime API.

File: SdlLogs/SdlLogs.py (hunk 12 above)
Comment (vilyukh) [reply SdlLogs/SdlLogs.py]: @LuxoftAKutsan Yes, flags are related to visibility status of columns (Date, Thread, Path) which are independent from each other.

File: SdlLogs/SdlLogs.py (hunk 4 above)
Comment (dkandyba) [reply SdlLogs/SdlLogs.py]: @LuxoftAKutsan  fixed in c95c56eb0fcc3c822bc445ccb58a40238b9d2ff2

File: SdlLogs/SdlLogs.py (hunk 5 above)
Comment (dkandyba) [reply SdlLogs/SdlLogs.py]: @LuxoftAKutsan  fixed in c95c56eb0fcc3c822bc445ccb58a40238b9d2ff2

File: SdlLogs/SdlLogs.py (hunk 6 above)
Comment (dkandyba) [reply SdlLogs/SdlLogs.py]: @LuxoftAKutsan  fixed in c95c56eb0fcc3c822bc445ccb58a40238b9d2ff2 
This argument needed for  function override

File: SdlLogs/SdlLogs.py (hunk 8 above)
Comment (dkandyba) [reply SdlLogs/SdlLogs.py]: @LuxoftAKutsan  fixed in c95c56eb0fcc3c822bc445ccb58a40238b9d2ff2

File: SdlLogs/SdlLogs.py (hunk 13 above)
Comment (dkandyba) [reply SdlLogs/SdlLogs.py]: @LuxoftAKutsan  fixed in c95c56eb0fcc3c822bc445ccb58a40238b9d2ff2

File: SdlLogs/SdlLogs.py (hunk 14 above)
Comment (dkandyba) [reply SdlLogs/SdlLogs.py]: @LuxoftAKutsan fixed in c95c56eb0fcc3c822bc445ccb58a40238b9d2ff2

File: SdlLogs/SdlLogs.sublime-color-scheme (hunk 15 above)
Comment (dkandyba) [reply SdlLogs/SdlLogs.sublime-color-scheme]: @LuxoftAKutsan  fixed in c95c56eb0fcc3c822bc445ccb58a40238b9d2ff2

File: SdlLogs/SdlLogs.py (hunk 7 above)
Comment (dkandyba) [reply SdlLogs/SdlLogs.py]: @LuxoftAKutsan fixed in 087ad4c948e9fb23a1428b129eb706c52696ddf7

File: README.md [hunk 16]
```diff
@@ -7,6 +7,8 @@ This plugin requires at least Sublime Text 3 v.3176 or higher for best performan
 ## Installing
 - Move directory "SdlLogs" to Sublime Packages folder.
 (Generally at /home/*USER*/.config/sublime-text-3/Packages)
+- Should install package for yaml
```
Comment (LuxoftAKutsan) [inline README.md]: @dkandyba extract list of dependencies to separate block

File: SdlLogs/SdlLogs.py [hunk 17]
```diff
@@ -3,157 +3,289 @@
...
+    ----------
+    self : class view 
+
+    Returns
+    ----------
+    str
+        Selected text
+    """
+    sel_text = self.view.sel()
+    sel_text = self.view.substr(sel_text[0])
```
Comment (LuxoftAKutsan) [inline SdlLogs/SdlLogs.py:116]: @dkandyba check if sel_text is empty

**Review by LuxoftAKutsan — APPROVED** [review]

File: SdlLogs/SdlLogs.py (hunk 17 above)
Comment (dkandyba) [reply SdlLogs/SdlLogs.py:116]: @LuxoftAKutsan not necessary check it. This works when cursor focus on view and if text is not selected will return empty string

File: README.md (hunk 16 above)
Comment (dkandyba) [reply README.md]: @LuxoftAKutsan fixed in def3edd1f30bd4e9f9281615b12f602163fb2ac4

**Pull request merged**